### ISSUE TITLE
[Epic] Phase 2: 制度文書ドメインでの PHOTON RepoRAG 精度検証（MVP 前提）

### ISSUE THREAD

**Issue by Kewton** [Kewton/photon-mlx#117]
## 目的

**MVP Phase 2（汎用化）** を再定義。Django / Pydantic などコードドメインの再検証はスキップし、**制度文書（日本語、法律・条文、4,228 ファイル）** で PHOTON RepoRAG の精度を確認する。

FastAPI で確立した改善（MT NC -41%、Turn 5-6 NC 0%、follow-up -34%）が **別ドメインでも再現するか** を実測し、MVP Phase 3（pip install 配布）へ進む判断材料とする。

## 対象コーパス

- **パス**: `/Users/maenokota/share/work/github_kewton/myWebData/markdowndb/institutional_documents`
- **規模**: 4,228 ファイル / 757,429 行
- **内容**: 日本語制度文書（法律、条文、報告書）— docling PDF 変換済
- **metadata.json** に source_url 保存

## 成功条件

| 指標 | 基準 | 計測 Issue | 実測 |
|------|------|-----------|------|
| 制度文書 Static NC | < 20% | #112 | **11.21%** ✅ (post-#125+#126 状態) |
| 制度文書 MT NC | < 15% | #113 | **6.94%** ✅ baseline / **11.39%** PHOTON 学習前 |
| 制度文書 follow-up latency | baseline -20%+ | #113 | **-44.9%** ✅ PHOTON working memory |
| **FastAPI 改善の再現率** | **70%+** | #116 | _未測_ |
| PHOTON Turn 5-6 NC | < 3% | #113（判定根拠）| **10.83%** PHOTON 学習前 → 仮説 C |

> **進捗 (2026-04-26 時点)**: Group 1 / 2 / 3 ✅ ほぼ完了。Phase 2 conditional 判定: **Turn 5-6 NC = 10.83% > 6% で仮説 C 該当 → 本格再学習 (#135) を active 化**。
> **3 重 handicap 状況**: (a) #125 + (c) #126 で組合せ解消 (NC 11.21%)、(b) は #133 で deferred。
> Group 5 follow-up: #123/#124/#127/#125/#126 closed、#133 (多言語 A/B) deferred、**#135 (本格再学習) を新規追加**。
> **PHOTON 速度優位 -44.9% 確認済**。再学習で精度を baseline 並みに回復できれば「速度 -45% × 精度同等」の優位確立。

### Group 1: Infrastructure（並列可能、先行着手）

| Issue | タイトル | 依存 | 工数 | 状態 |
|-------|---------|------|------|------|
| #109 | feat(ingestion): markdown chunker + symbol graph conditional skip | なし | 3 日 | ✅ merged |
| #110 | feat(eval): 制度文書 eval set 自動生成 | なし | 4 日 | ✅ merged |
| #111 | feat(retrieval): query_expansion と noise_patterns の汎用化 | なし | 3 日 | ✅ merged |

### Group 2: Integration（Group 1 後）

| Issue | タイトル | 依存 | 工数 | 状態 |
|-------|---------|------|------|------|
| #112 | feat(configs): 制度文書プロファイル + index + baseline Static eval | #109, #110, #111 | 3 日 | ✅ merged (NC 11.21%) |

### Group 3: Measurement & Tuning（Group 2 後、一部並列）

| Issue | タイトル | 依存 | 工数 | 状態 |
|-------|---------|------|------|------|
| #113 | **measure: 現行 PHOTON で MT eval 実測（学習要否の判定根拠）** | #112 | 2 日（クリティカルパス）| ✅ merged (Turn 5-6 NC 10.83%, latency -44.9%) |
| #114 | feat(retrieval): 多言語 embedding / reranker A/B | #112 | 4 日 | ✅ partial (#132 guard test) → A/B は #133 deferred |

### Group 4: UI & Final（Group 3 後）

| Issue | タイトル | 依存 | 工数 | 状態 |
|-------|---------|------|------|------|
| #115 | feat(app): wizard domain template + 日本語 prompt | #112, #114 | 2 日 | ⏸️ 未着手 |
| #116 | docs: Phase 2 完了レポート（FastAPI vs 制度文書 再現率）| 全 Issue | 1 日 | ⏸️ 全完了後 |

### Conditional（#113 の結果で新規登録）— **本格再学習 active**

| シナリオ | Issue 内容 | 工数 | 状態 |
|---------|----------|------|------|
| Turn 5-6 NC < 3% | なし（学習不要）| 0 日 | ❌ 棄却 (10.83%) |
| Turn 5-6 NC 3-6% | 軽量 fine-tune（5K step、JP 混合）| 3 日 | ❌ 棄却 (10.83%) |
| **Turn 5-6 NC > 6%** | **本格再学習（10-20K step、JP 50%+）** | **5 日** | ✅ **#135 (active)** |

### Group 5: Follow-up（#112 完了後、handicap 是正・運用整備）

#112 baseline 実測の 3 重 handicap (a)/(b)/(c) 是正、CI/運用整備、再現性向上のための後続 Issue。Group 3 (#114) と並列実行可能で、特に **#125 は #114 の prerequisite**（E5 prefix 適用後の variant 比較が正確）。

| Issue | タイトル | 依存 | 工数 | handicap | 状態 |
|-------|---------|------|------|---------|------|
| #125 | feat(indexing): E5 embedding で query:/passage: prefix を付与 | #112 | 1-2 時間 | (a) | ✅ merged (PR #130, regression は #126 で解消) |
| #126 | tune(ingestion): 日本語 markdown chunker max_chars 実験 | #112 | 半日 | (c) | ✅ merged (PR #131, NC 11.21% へ回復) |
| #124 | chore(ci): baseline_eval log glob を fastapi_fastapi pattern に絞る | #112 | 30 分 | — | ✅ merged (PR #128) |
| #127 | feat(scripts): aggregate_institutional_baseline.py 新設 | #112 | 1-2 時間 | — | ✅ merged (PR #129) |
| #133 | feat(retrieval): 多言語 reranker A/B 実機評価 (#114 follow-up) | #112 | 3-4 時間 | (b) | 🟡 open (deferred、別セッションで実施) |

> handicap (b) 多言語 cross-encoder reranker は **#114 で guard test (#132) のみ commit**、A/B 実機評価は **#133** に分離継承。

> #110 follow-up の **#123 (fix: institutional generator defensive filter)** は #117 配下ではなく単独 fix Issue として扱った（merged, direct commit `adafd08`）。

### Group 6: Conditional Re-training（#113 結果で active 化）

| Issue | タイトル | 依存 | 工数 | 状態 |
|-------|---------|------|------|------|
| **#135** | feat(training): PHOTON 本格再学習 — JP 50%+ 混合 (10-20K step) | #113 / #126 / #125 | **5 日** | 🟡 **open / active** |

## 並列実行プラン

```
Week 1:
  並列 → #109 (chunker) + #110 (eval set) + #111 (query/noise)
        完了
        ↓
  直列 → #112 (config + baseline eval)

Week 2:
  並列 → #113 (PHOTON MT eval) + #114 (embedding/reranker A/B)
        完了
        ↓
  判定 → 再学習要否（conditional issue 作成 or なし）
        ↓
  並列 → #115 (wizard) + (必要なら) 学習 Issue

Week 3:
  直列 → #116 (Phase 2 完了レポート)
        ↓
  判定 → MVP Phase 3 進行可否
```

想定総工数: **~18 日**（3-4 週間、1 人 full time、Group 1 並列で圧縮）

## タイムライン

- Week 1: Group 1 並列実行 → Group 2 統合
- Week 2: 計測と tuning（判定根拠確定）
- Week 3: UI・レポート・Phase 2 判定

## リスク

| リスク | 影響 | 対策 |
|--------|------|------|
| PHOTON が日本語で encoding 崩れる | MVP 遅延 | #113 で早期判定、fallback として fine-tune |
| 制度文書の構造が特殊（附則、改正履歴等）| chunk 品質低下 | markdown_chunker で header 優先分割 |
| eval set の人手検証コスト大 | 品質 vs 速度トレードオフ | LLM 自動生成 + 人手 20% サンプル検証 |
| 日本語 reranker がメモリ不足 | ハード制約 | `multilingual-e5-small` + 軽量 reranker のフォールバック |

## 関連

- Epic #93 Wave 6（完了、2026-04-23 クローズ）
- Issue #81 Static NC < 15% Epic（本 Epic で #88 harness が制度文書にも適用可能）
- Issue #49 PHOTON Medium 1B（本 Epic で再学習必要性が判明すれば触発要素）

## Next Action

**現状 (2026-04-26)**: Group 1 / 2 / 3 / 5 ✅ ほぼ完了。**Phase 2 のクリティカルパスは #135 本格再学習**。

### 完了済 (2026-04-25 → 04-26)
- Group 1 #109/#110/#111 ✅、Group 2 #112 ✅、Group 3 #113/#114 partial ✅
- Group 5 #123/#124/#125/#126/#127 ✅ closed
- Phase 2 conditional 「本格再学習」シナリオの判定根拠 (#113 で確定)

### 推奨着手順

1. **#135 (本格再学習)** クリティカルパス、5 日工数。再学習 → 再 MT eval → MVP Phase 2 達成判定
2. **#133 (多言語 reranker A/B)** #135 と並列可能、deferred 状態から再開 (3-4 時間)
3. **#115 (Streamlit wizard)** Group 4、#135 完了後に着手（再学習 PHOTON を UI で使えるようにする）
4. **#116 (Phase 2 完了レポート)** 全完了後に着手、FastAPI vs 制度文書 再現率比較

### 現状の優位ポイント
- baseline NC 11.21% で reasonable range 達成
- PHOTON working memory **latency -44.9%** 確認済 → 再学習で精度 baseline 並みに戻れば「速度 -45% × 精度同等」確立
- 3 重 handicap のうち (a)+(c) 解消済、(b) は #133 で deferred

**Comment by Kewton** [comment]
## 🎉 Phase 2 完全完了 — Epic Close

### 完了 Issue (Phase 2 全 12 件)

| Issue | 内容 | 解消 PR | 状態 |
|---|---|---|---|
| #109 | (Phase 2 初期 Issue) | (既存) | ✅ |
| #110 | (Phase 2 初期 Issue) | (既存) | ✅ |
| #111 | (Phase 2 初期 Issue) | (既存) | ✅ |
| #112 | Static eval baseline | (既存) | ✅ |
| #113 | PHOTON MT eval (baseline) | (既存) | ✅ |
| #114 | Multilingual reranker A/B | (既存) | ✅ |
| **#115** | wizard + JP prompt | #149 | ✅ |
| **#116** | Phase 2 完了レポート | #158 | ✅ |
| **#137** | 5-variant retrieval A/B + V4 採用 | #142 | ✅ |
| **#138** | tokenizer mismatch fix | #141 | ✅ |
| **#139** | Stub/Mock audit | #147 | ✅ |
| **#140** | Codex review process 改善 | #146 | ✅ |
| **#148** | re-baseline (Phase A0+A+A.3) | #150-153 | ✅ |
| **#154** | citation grader + cross-repo fix | #155 | ✅ |
| **#135** | PHOTON 本格再学習 | #157 | ✅ |

### 達成事実 (refusal-aware 判定)

| 受入条件 | 結果 |
|---|---|
| 制度文書 MT NC < 15% | ✅ **0.00%** |
| Follow-up latency baseline -20%+ | ✅ **-37.7%** |
| FastAPI 改善の 70%+ 再現 | ✅ **Indicator A 100% / B 110%** |

### 採用 PHOTON checkpoint

`photon_institutional_retrain_20260428/step_003000` (val_loss 0.4777、JP:0.7/EN:0.3 mix、累計 step 3000)

### 主要発見 (Phase 2 中)

1. **5 つの critical bug を発見・修正**: S7-001、tokenizer mismatch、vocab padding、num_heads schema drift、cross-repo leakage
2. **訓練データ仮説確定**: 訓練 domain (TS/JP) で PHOTON 圧勝、commandmate eval で実証
3. **真の PHOTON 性能**: random-init era の幻想を解消、refusal-aware で 0% NC 達成
4. **eval pipeline 健全化**: refusal-aware grader、repo isolation 完備

### 残 follow-up (Phase 3 着手前推奨)

- #156 (計測 bug、is_refusal 出力欠落)
- #143 (Qwen nondeterminism)
- #144 (V2 ruri build)
- #145 (real-weight integration test)

### 次のフェーズ

**Phase 3 (MVP 配布) Conditional Go** — `reports/phase2_cross_domain_validation.md` で勧告済。Condition: #156 解消後、Phase 3 着手可。

Phase 2 集約 Epic として本 Issue を close。
